src/allocator.cpp

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+#include "allocator.h"
+#include "utils.h"
+
+namespace diskann
+{
+
+template <class T> T *Allocator<T>::allocate(std::size_t count)
```

Comment:
why do need an allocator class? Can this be an internal detail of memory_manager class?

Reply:
The Allocator class is planned to use in std::vector, std::unordered_map .etc to track their memory usage